specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,573 @@
+
+    counts : astropy.Quantity array
+        Counts histogram for the X-ray spectrum
+
+    exposure : float
+        Exposure time for the dataset
+
+    arf : specutils.AreaResponse or string, default None
+        Strings will be passed to AreaResponse.__init__
+        All other input types are stored as arf attribute
+
+    rmf : specutils.ResponseMatrix or string, default None
+        Strings will be passed to ResponseMatrix.__init__
+        All other input types are stored as rmf attribute
+
+    rest_value : astropy.units.Quantity, default 0 Angstrom
+        See Spectrum1D rest_value input
+
+    Attributes
+    ----------
+    Inherits from Spectrum1D object.
+    The following inputs are stored as additional attributes.
+
+    bin_lo
+
+    bin_hi
+
+    counts
+
+    exposure
+
+    arf
+
+    rmf
+    """
+    def __init__(self, bin_lo, bin_hi, bin_unit, counts, exposure,
+                 arf=None, rmf=None, rest_value=0.0 * u.angstrom, **kwargs):
+        try:
+            axis_unit = u.Unit(bin_unit)
+        except ValueError:
+            axis_unit = _unit_parser(bin_unit)
+
+        bin_mid = 0.5 * (bin_lo + bin_hi) * axis_unit
+        Spectrum1D.__init__(self, spectral_axis=bin_mid, flux=counts,
+                            rest_value=rest_value, **kwargs)
+
+        self.bin_lo = bin_lo
+        self.bin_hi = bin_hi
+        self.exposure = exposure
+        self.assign_rmf(rmf)
+        self.assign_arf(arf)
+
+    # Convenience function for Xray people
+    @property
+    def counts(self):
+        return self.flux
+
+    def assign_arf(self, arf_inp):
+        """
+        Assign an area response file (ARF) object to the XraySpectrum1D object
+
+        Input
+        -----
+        arf_inp : string
+            File name for the area response file (FITS file)
+
+        Returns
+        -------
+        Modifies the XraySpectrum1D.arf attribute
+        """
+        if isinstance(arf_inp, str):
+            self.arf = AreaResponse(arf_inp)
+        else:
+            self.arf = arf_inp
+
+    def assign_rmf(self, rmf_inp):
+        """
+        Assign a response matrix file (RMF) object to the XraySpectrum1D object
+
+        Input
+        -----
+        rmf_inp : string
+            File name for the response matrix file (FITS file)
+
+        Returns
+        -------
+        Modifies the XraySpectrum1D.rmf attribute
+        """
+        if isinstance(rmf_inp, str):
+            self.rmf = ResponseMatrix(rmf_inp)
+        else:
+            self.rmf = rmf_inp
+        return
+
+    def apply_response(self, mflux, exposure=None):
+        """
+        Given a model flux spectrum, apply the response. In cases where the
+        spectrum has both an area response file (ARF) and a response matrix
+        file (RMF), apply both. Otherwise, apply whatever response is in RMF.
+
+        The model flux spectrum *must* be created using the same units and
+        bins as in the ARF (where the ARF exists)!
```

Comment:
I wonder if there is a way to enforce that somehow. Seems like a classic point of failure.